cli/flags/common.go

```diff
@@ -45,13 +45,14 @@ var (
 
 // CommonOptions are options common to both the client and the daemon.
 type CommonOptions struct {
-	Debug      bool
-	Hosts      []string
-	LogLevel   string
-	TLS        bool
-	TLSVerify  bool
-	TLSOptions *tlsconfig.Options
-	Context    string
+	Debug       bool
+	Hosts       []string
+	LogLevel    string
+	TLS         bool
+	TLSVerify   bool
+	TLSOptions  *tlsconfig.Options
+	Context     string
+	HttpHeaders []string
```

Comment:
w.r.t. the above (parsing) in mind I'm thinking if perhaps the value parsing should be part of a custom type (so that this would be an (e.g.) `opts.HttpHeadersOpt` with its own validation function, similar fo [`opts.ListOpts`](https://github.com/docker/cli/blob/c780f7c4abaf67034ecfaa0611e03695cf9e4a3e/opts/opts.go#L22) and [`opts. NamedListOpts`](https://github.com/docker/cli/blob/c780f7c4abaf67034ecfaa0611e03695cf9e4a3e/opts/opts.go#L132)

Thinking of which (sorry, bit hazy on that one), but I _think_ the `NamedListOpts` and `NamedOpts` types were used for command-line flags that also have an equivalent in `~/.docker/config.json`. Which (in this case) could be considered the case (the `--http-headers` being an equivalent for the `HttpHeaders` option in the config file)

(Hope this helps, but happy to do some further digging in this)

Reply:
Basically, you want to use `opts.NamedMapOpts` and according validator, right?